optimum/habana/transformers/trainer.py

```diff
@@ -226,6 +226,40 @@ def __init__(
             "ignore", message="User provided device_type of 'cuda', but CUDA is not available. Disabling"
         )
 
+        if train_dataset is not None:
+            import inspect
+
+            from optimum.habana.fx import GeluToFusedGelu, symbolic_trace
+
+            transformation = GeluToFusedGelu()
+
+            input_names_for_tracing = list(train_dataset.features.keys())
+            for i, name in enumerate(input_names_for_tracing[:]):
+                if name == "image":
+                    input_names_for_tracing[i] = "pixel_values"
+
+            training_traced = symbolic_trace(
+                model,
+                input_names=input_names_for_tracing,
+            )
+            transformed_training_traced = transformation(training_traced)
+            self.training_fwd = transformed_training_traced.forward
+
+            input_names_for_tracing = []
+            sig = inspect.signature(model.forward)
+            for name in list(eval_dataset.features.keys()):
```

Comment:
what happens if `eval_dataset` is `None`, which means user not set `--do_eval`

Reply:
Yes I will add a `if self.eval_dataset is not None` statement as I did above with the training dataset. I was only testing so I have not done it yet but I will.

Reply:
One solution as well is to create one batch of data and use it to infer the input names. But this might be more costly.